### PR TITLE
Convert the skydns rc to use v1beta3 and add a health check to it

### LINKED ISSUE #4852: Need a health check for DNS pod
I have observed the DNS pod sometimes getting stuck.  Need a health-check that tries to do a lookup or something like that.

### PR DESCRIPTION
**Pull request by a-robinson**
DO NOT MERGE YET - exec probing is currently broken as far as I can tell (#7618), so this will cause the skydns container to be restarted every 30-ish seconds if it's merged before that issue is fixed.

Fixes #4852

@vishh

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Thanks. This helps with https://github.com/GoogleCloudPlatform/kubernetes/issues/6584

**Comment by vishh** [comment]
@a-robinson: Kubelet exec has been fixed on HEAD. Can you rebase this PR and test it once more?

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 1]
```diff
@@ -1,63 +1,66 @@
...
+    metadata:
+      labels:
+        k8s-app: kube-dns
+        kubernetes.io/cluster-service: "true"
+        name: kube-dns
+    spec:
+      containers:
+      - name: etcd
+        image: gcr.io/google_containers/etcd:2.0.9
+        args:
```
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: This should be command

**Comment by thockin** [comment]
1 nit, LGTM

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 1 above)
Comment (a-robinson) [inline cluster/addons/dns/skydns-rc.yaml.in]: Thanks, I wonder how it worked when I tested it like that...

**Comment by a-robinson** [comment]
Re-tested after rebasing on top of CJ's changes. Before kube-push'ing the new code with the exec fix, the skydns container restarted every 30-35 seconds. After kube-push'ing the new code, it doesn't restart.

**Comment by vishh** [comment]
LGTM

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 1 above)
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: It works as args but is not very self describing. Arguably you could do
command and args but meh.

**Comment by thockin** [comment]
LGTM

**Pull request merged**